File: python_Programmers/1단계/행렬의덧셈.py
import numpy as np

# 내 풀이
def solution(arr1, arr2):
    arr1 = np.array(arr1)
    arr2 = np.array(arr2)
    return (arr1 + arr2).tolist()

# 모범 풀이
def solution2(arr1,arr2):
    for a, b in zip(arr1, arr2):
        print('a : ', a)
        print('b : ', b)
        for c, d in zip(a, b):
            print('c : ', c)
            print('d : ', d)

    answer = [[c + d for c, d in zip(a, b)] for a, b in zip(arr1,arr2)]
    return answer

arr1 = [[1,2],[2,3]]
arr2 = [[3,4],[5,6]]
print(solution(arr1, arr2))
print(solution2(arr1, arr2))

File: Java_Programmers/1단계/src/행렬의덧셈.java
import java.util.Arrays;

public class 행렬의덧셈 {
    public int[][] solution(int[][] arr1, int[][] arr2) {
        int size = arr1.length;
        int size2 = arr1[0].length;
        int[][] answer = new int[size][size2];
        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size2; j++) {
                answer[i][j] = arr1[i][j] + arr2[i][j];
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        행렬의덧셈 sol = new 행렬의덧셈();
//        int[][] arr1 = {{1, 2}, {2, 3}};
        int[][] arr1 = {{1}, {2}};
//        int[][] arr2 = {{3, 4}, {5, 6}};
        int[][] arr2 = {{3}, {4}};
        System.out.println(Arrays.deepToString(sol.solution(arr1, arr2)));
    }
}
